src/core/server/saved_objects/migrationsv2/model.test.ts

```diff
@@ -863,9 +863,10 @@ describe('migrations v2 model', () => {
         });
         const newState = model(testState, res) as FatalState;
         expect(newState.controlState).toBe('FATAL');
-        expect(newState.reason).toMatchInlineSnapshot(
-          `"Migrations failed. Reason: Corrupt saved object documents: a:b. To allow migrations to proceed, please delete these documents."`
-        );
+        expect(newState.reason).toMatchInlineSnapshot(`
+          "Migrations failed. Reason: 1 corrupt saved object documents were found: a:b
+          To allow migrations to proceed, please delete or fix these documents."
```

Comment:
So, I did not handle singular/plural in the message, as it didn't really feel that important, but if we really want it, I could do it.